File: Calculadora31Oct2019/app/src/main/java/com/example/calculadora31oct2019/MainActivity.java
package com.example.calculadora31oct2019;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private Button btsuma;
    private Button btresta;
    private Button btmulti;
    private Button btdiv;
    private EditText et1;
    private EditText et2;
    private TextView tvres;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btsuma  = (Button) findViewById(R.id.btsuma);
        btresta = (Button) findViewById(R.id.btresta);
        btmulti = (Button) findViewById(R.id.btmulti);
        btdiv   = (Button) findViewById(R.id.btdiv);
        et1     = (EditText) findViewById(R.id.et1);
        et2     = (EditText) findViewById(R.id.et2);
        tvres   = (TextView) findViewById(R.id.tvres);
        btsuma.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {               //-- Listener para botón suma.
                double res;
                res=(Double.parseDouble(et1.getText().toString()))+(Double.parseDouble(et2.getText().toString()));
                tvres.setText(Double.toString(res));
            }
        });
        btresta.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {               //-- Listener para botón resta.
                double res;
                res=(Double.parseDouble(et1.getText().toString()))-(Double.parseDouble(et2.getText().toString()));
                tvres.setText(Double.toString(res));
            }
        });
        btmulti.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {               //-- Listener para botón multiplicación.
                double res;
                res=(Double.parseDouble(et1.getText().toString()))*(Double.parseDouble(et2.getText().toString()));
                tvres.setText(Double.toString(res));
            }
        });
        btdiv.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {               //-- Listener para botón división.
                double res;
                res=(Double.parseDouble(et1.getText().toString()))/(Double.parseDouble(et2.getText().toString()));
                tvres.setText(Double.toString(res));
            }
        });
    }
}

File: Calculadora/app/src/main/java/com/example/calculadora/MainActivity.kt
package com.example.calculadora

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.View.OnClickListener
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(),View.OnClickListener {
    override fun onClick(v: View?) {
        //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
        when(v?.id){
            R.id.btnAdd ->{
                if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                    txtResult.text = "Result: ${etxData1.text.toString().toInt() + etxData2.text.toString().toInt()}"
                }else{
                    showMessage("No Data")
                }
            }
            R.id.btnSus ->{
                if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                    txtResult.text = "Result: ${etxData1.text.toString().toInt() - etxData2.text.toString().toInt()}"
                }else{
                    showMessage("No Data")
                }
            }
            R.id.btnMul ->{
                if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                    txtResult.text = "Result: ${etxData1.text.toString().toInt() * etxData2.text.toString().toInt()}"
                }else{
                    showMessage("No Data")
                }
            }
            R.id.btnDiv ->{ if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                if (etxData2.text.toString().toInt() == 0) {
                    showMessage("Impossible Division")
                } else {
                    txtResult.text =
                        "Result: ${etxData1.text.toString().toFloat() / etxData2.text.toString().toFloat()}"
                }
            }else showMessage("No Data")
            }

        }
    }

    @SuppressLint("SetTextI18n")


/*    object MyOnClick: OnClickListener {
        override fun onClick(v: View?) {
            //TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
            when(v?.id){
                R.id.btnAdd ->{

                }
                R.id.btnSus ->{ }
                R.id.btnMul ->{ }
                R.id.btnDiv ->{ }

            }
        }
    }*/


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnAdd.setOnClickListener(this)
        btnSus.setOnClickListener(this)
        btnMul.setOnClickListener(this)
        btnDiv.setOnClickListener(this)



/*        btnAdd.setOnClickListener {

            if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                txtResult.text = "Result: ${etxData1.text.toString().toInt() + etxData2.text.toString().toInt()}"
            }else{
                showMessage("No Data")
            }
        }

        btnSus.setOnClickListener {

            if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                txtResult.text = "Result: ${etxData1.text.toString().toInt() - etxData2.text.toString().toInt()}"
            }else{
                showMessage("No Data")
            }
        }

        btnMul.setOnClickListener {

            if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                txtResult.text = "Result: ${etxData1.text.toString().toInt() * etxData2.text.toString().toInt()}"
            }else{
                showMessage("No Data")
            }
        }

        btnDiv.setOnClickListener {
            if(!(etxData1.text.toString().isBlank()||etxData2.text.toString().isBlank())) {
                if (etxData2.text.toString().toInt() == 0) {
                    showMessage("Impossible Division")
                } else {
                    txtResult.text =
                        "Result: ${etxData1.text.toString().toFloat() / etxData2.text.toString().toFloat()}"
                }
            }else{
                showMessage("No Data")
            }

        }*/
    } // end of onCreate

    private fun showMessage(s: String) {
        Toast.makeText(this,s,Toast.LENGTH_SHORT).show()
    }
}